components/raftstore/src/store/fsm/apply.rs

```diff
@@ -5491,6 +5550,102 @@ mod tests {
         });
     }
 
+    #[test]
+    fn test_exec_observer() {
```

Comment:
add some comments for the test steps and the logic to verify.

Reply:
Will do after pre_exec is merged